quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -1096,21 +1096,47 @@ private static void generateDomainAccessorsForShadowSources(IndexView indexView,
             var sources = shadowSources.value().asStringArray();
             for (var source : sources) {
                 var currentType = rootType;
+                java.lang.reflect.Type currentGenericType = rootType;
                 var parts = source.split(RootVariableSource.MEMBER_SEPERATOR_REGEX);
+                boolean isAfterCollection = false;
+                String collectionName = "";
                 for (var part : parts) {
-                    var memberName = part.endsWith(RootVariableSource.COLLECTION_REFERENCE_SUFFIX)
+                    var isCollection = part.endsWith(RootVariableSource.COLLECTION_REFERENCE_SUFFIX);
```

Comment:
Do we have the path parsing code separately in Core and in Quarkus?
Preferrably, we'd reuse the parsing logic from Solver.